src/nfa/builder.cc

```diff
@@ -204,3 +207,36 @@ Nfa Builder::create_sigma_star_nfa(Mata::Alphabet* alphabet) {
     }
     return nfa;
 }
+
+Nfa Builder::parse_from_mata(std::istream& nfa_stream) {
+    const std::string nfa_str = "NFA";
+    Parser::Parsed parsed{ Parser::parse_mf(nfa_stream) };
+    if (parsed.size() != 1) {
+        throw std::runtime_error("The number of sections in the input file is not 1\n");
+    }
+    if (parsed[0].type.compare(0, nfa_str.length(), nfa_str) != 0) {
+        throw std::runtime_error("The type of input automaton is not NFA\n");
```

Comment:
I would suggest to extend both of the exception message to include the actual value of the parameters (i.e. number of sections and type of the automaton).

Reply:
Added. Good suggestion.